radarpipeline/io/reader.py

```diff
@@ -255,8 +255,12 @@ def _get_field(self, data_type: Union[str, Dict]) -> Any:
         Any
             A Spark data type
         """
-
-        if type(data_type) is str and data_type in self.data_type_mapping:
+        if type(data_type) is list:
+            if len(data_type) == 1:
```

Comment:
```suggestion
            if len(data_type) == 1 and data_type[0] in self.data_type_mapping:
```

I think we should also check if the first data type of a list is in the datatype mapping. If it is not, it would go to the `else` condition.